PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx

```diff
@@ -427,40 +438,45 @@ struct HfTaskFlowCharmHadrons {
                const float occupancy,
                const o2::hf_evsel::HfCollisionRejectionMask hfevselflag)
   {
+    auto hSparse = registry.get<THnSparse>(HIST("hSparseFlowCharm"));
+
+    std::array<double, 32> values{};
+    int n = 0;
+
+    values[n++] = mass;
+    values[n++] = pt;
+    values[n++] = cent;
+    values[n++] = sp;
+
+    if (storeEP) {
+      values[n++] = cosNPhi;
+      values[n++] = sinNPhi;
+      values[n++] = cosDeltaPhi;
+    }
+    if (storeMl) {
+      values[n++] = outputMl[0];
+      values[n++] = outputMl[1];
+    }
+    if (storeCandEta) {
+      values[n++] = eta;
+    }
```

Comment:
This is very bug-prone. The positions of elements depend on external switches.

Reply:
Thanks for the comment, I see your point. In this implementation the element positions are controlled by the same switches and in the same order as in the axis definition